.github/workflows/run-tests.yml

```
@@ -29,7 +55,29 @@ env:
   BUILDKITE_ANALYTICS_TOKEN: ${{ secrets.BUILDKITE_ANALYTICS_TOKEN }}
 
 jobs:
+
+  cache-docker-images:
+    if: ${{ inputs.run_single_functional_test != true }}
+    name: Cache Docker images
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [ubuntu-20.04]
+    runs-on: ${{ matrix.runs-on }}
+    steps:
+      - uses: ScribeMD/docker-cache@0.3.7
+        with:
+          key: docker-${{ runner.os }}-${{ hashFiles(env.DOCKER_COMPOSE_FILE) }}
+
+      - uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          ref: ${{ env.COMMIT }}
+
+      - run: docker compose -f ${{ env.DOCKER_COMPOSE_FILE }} pull
+
   misc-checks:
+    if: ${{ inputs.run_single_functional_test != true }}
```

Boolean inputs becomes strings.
```suggestion
    if: ${{ github.event.inputs.run_single_functional_test != 'true' }}
```